crates/bevy_ecs/src/event.rs

```diff
@@ -448,8 +495,12 @@ impl<E: Event> Events<E> {
     }
 
     /// A system that calls [`Events::update`] once per frame.
-    pub fn update_system(mut events: ResMut<Self>) {
-        events.update();
+    pub fn update_system(mut events: ResMut<Self>, settings: Option<Res<EventSettings<E>>>) {
+        if let Some(settings) = settings {
+            events.update(&settings);
+        } else {
+            events.update(&Default::default());
```

Comment:
Does `default()` works here (and below)? Or we need the extra `Default::` qualification?